pulp_puppet_plugins/pulp_puppet/plugins/importers/importer.py

```diff
@@ -80,9 +81,21 @@ def upload_unit(self, repo, type_id, unit_key, metadata, file_path, conduit, con
         try:
             report = upload.handle_uploaded_unit(repo, type_id, unit_key, metadata, file_path,
                                                  conduit)
+
```

Comment:
I'm not very familiar with these exception types, but I think they each should inherit from PulpCodedException and be instantiated similar to option 3.